File: EASY/PALINDROME NUMBER/palindrome_num.py
class Solution:
    def isPalindrome(self, x) -> bool:
        r = 0
        a = x
        if x < 0:
            return False
        while x!=0:
            digit = x % 10
            r = r * 10 + digit
            x //= 10
        return (r == a)
    
ob = Solution()
x = ob.isPalindrome(121)
y = ob.isPalindrome(-121)
z = ob.isPalindrome(10)
print(x)
print(y)
print(z)

File: EASY/PALINDROME NUMBER/palindromeNum.java
class palindromeNum{
    public boolean isPalindrome(int x){
        int rev = 0, digit, a=x;
        if(x<0){
            return false;
        }
        while (x != 0){
            digit = x % 10;
            rev = rev * 10 + digit;
            x /= 10;
        }
        return (a == rev);
    }
    public static void main(String[] args) {
        palindromeNum p = new palindromeNum();
        boolean x = p.isPalindrome(121);
        boolean y = p.isPalindrome(-121);
        boolean z = p.isPalindrome(10);
        System.err.println(x);
        System.err.println(y);
        System.err.println(z);
    }
}